packages/salesforcedx-vscode-apex-replay-debugger/src/adapter/debugConfigurationProvider.ts

```diff
@@ -40,13 +40,21 @@ export class DebugConfigurationProvider
     config.name = config.name || nls.localize('config_name_text');
     config.type = config.type || DEBUGGER_TYPE;
     config.request = config.request || DEBUGGER_LAUNCH_TYPE;
+    config.projectBreakpointFilePath = '${command:sfdx.updateBreakpoint}';
```

Comment:
@lcampos - I think this is they 🔑 idea here. Could you elaborate more on why `${command:sfdx.updateBreakpoint}` is guaranteed to invoke the command? I think it's because of https://code.visualstudio.com/docs/extensions/example-debuggers#_anatomy-of-the-packagejson-of-a-debug-extension

> The variables contribution binds 'variables' to 'commands'. These variables can be used in the launch configuration using the ${command:Xxxx} syntax and the variables are substituted by the value returned from the bound command when a debug session is started.

1. Are there any limits to how many configuration variables can be "bound" by such dynamic commands?
1. Do the "bound" commands execute in order (or in parallel)?
1. What happens if one of the "bound" command fails? Does the rest of the debugger sequence continue trying to invoke other "bound" commands? Or it bails?

Reply:
Also, if we put this in a config, would it also show up when the user is typing the configuration? I think VS Code will try to provide some auto-completion for you when you are typing out the debugger configuration.

In the grand scheme of things this is not a stopper. But it would be good if we could prevent the user from overriding the `projectBreakpointFilePath` key in the Debug Launch Configuration so that nothing bad will happen.

Reply:
I didn't find much documentation on it so the following is based off of code and comments on threads for the feature.

1. Looking at the code that collects and executes the variables used in the launch config there are no limits to how many commands can be included. The following does apply; the command is only executed once if it's declared multiple times on the same launch configuration, the debug session is canceled if the command errors or is itself canceled (e.g. if the command provides a UI to do so), if the variable can't be mapped to a registered command the debug session execution is canceled. 

Here are some test for multiple commands per configuration https://github.com/Microsoft/vscode/blob/90171dab977e4cea933a01824ed0313d86eccdc8/src/vs/workbench/services/configurationResolver/test/electron-browser/configurationResolverService.test.ts

2. Commands get executed sequentially https://github.com/Microsoft/vscode/blob/2d452e0a7e981a675a59617e296b51de5c40bbb7/src/vs/workbench/services/configurationResolver/electron-browser/configurationResolverService.ts#L109

3. If one of the commands fails then the debug session is also canceled.

Here are the threads on when the feature was added (https://github.com/Microsoft/vscode/issues/6569, https://github.com/Microsoft/vscode/issues/7026). It's been refactored since then but the specs still apply (look at links in answer 2).

Reply:
@vazexqi I haven't test if this new field will be available for auto-completion by the user, I'll get back to you on that (good feedback).

Reply:
I've confirmed that this new property is not available to the end user since it is not defined on the extension's `package.json`. In case they add it to their config it will be labeled as invalid and that field will be ignored when running the replay debugger.

Reply:
@JimSuplizio that seem like a good change. I remember trying to do the async call here but don't exactly remember what issue I ran into. Let me try again and do some testing.

Reply:
Thanks for the suggestion @JimSuplizio. Keep us posted @lcampos on whether the suggested approach works — or, if it doesn't, why it can't work for our case.

Reply:
I've confirmed that @JimSuplizio suggestion works. The reason I was initially having issues with this was that I was trying to directly convert resolveDebugConfiguration into an async method and typescript was throwing errors. Moving the configuration to an async method that is then resolved on resolveDebugConfiguration's return does the trick :)

I'll be closing this PR and opening a new one with all the changes discussed here.

cc @vazexqi

Reply:
Excellent!